libraries/TuneScope/TS_init.js

```diff
@@ -16,6 +16,8 @@ for (var i = 0; i <= 127; i++) {
     tempMidiFreqs[note] = 440 * Math.pow(2, (i - 69)/12)
 }
 
+window.currentNote = ""
```

Comment:
Sigh. I'm not really happy with extensions spamming the global namespace, and neither can you. Remember that these changes will stick around even after the user makes a new project that doesn't use TuneScope or opens another project in the same session. Extensions should strive to be ... extensions and at least attempt some kind of modularity. Actually, keeping track of state is the *one* use case that makes sense for the "make variables" library. You could initialize a TuneScope "_current note" global variable (notice the underscore so it wouldn't conflict with anything the user might create - and just use that inside your custom block. But I realize that other extensions also suffer from this anti-pattern of just spamming the global namespace with their variable, hence my "sigh" - so I guess it's okay if you do so, too. I was hoping - perhaps unreasonably so - that you might be smarter than some of the others. I'm still going to pull this, but I'd like you to consider daring to become more of a systems architect than a duck-tape hacker ;)